tests/integration/targets/mso_schema_template_bd/tasks/main.yml

```diff
@@ -883,6 +912,277 @@
+          version: 1
+    state: present
+  register: nm_bd_new_subnet_options
+
+- name: Verify subnets in nm_bd_new_subnet_options
+  assert:
+    that:
+    - nm_bd_new_subnet_options is changed
+    - nm_bd_new_subnet_options.current.name == 'ansible_test_multiple_dhcp'
+    - nm_bd_new_subnet_options.current.vrfRef.templateName == "Template1"
+    - nm_bd_new_subnet_options.current.vrfRef.vrfName == "VRF"
+    - nm_bd_new_subnet_options.current.intersiteBumTrafficAllow == true
+    - nm_bd_new_subnet_options.current.optimizeWanBandwidth == false
+    - nm_bd_new_subnet_options.current.l2Stretch == true
+    - nm_bd_new_subnet_options.current.l2UnknownUnicast == "flood"
+    - nm_bd_new_subnet_options.current.unkMcastAct == "flood"
+    - nm_bd_new_subnet_options.current.multiDstPktAct == "drop"
+    - nm_bd_new_subnet_options.current.l3MCast == false
+    - nm_bd_new_subnet_options.current.arpFlood == true
+    - nm_bd_new_subnet_options.current.v6unkMcastAct == "flood"
+    - nm_bd_new_subnet_options.current.vmac == "00:00:5E:00:01:3C"
+    - nm_bd_new_subnet_options.current.unicastRouting == true
+    - nm_bd_new_subnet_options.current.subnets[1].primary == false
+    - nm_bd_new_subnet_options.current.subnets[0].virtual == false
+    - nm_bd_new_subnet_options.current.subnets[2].primary == true
+    - nm_bd_new_subnet_options.current.subnets[2].virtual == true
+  when: version.current.version is version('3.1.1g', '>')
+
+# Change BD with new options for mso version > 3.1.1g
+- name: Try Changing bd with another subnet to primary IP
+  mso_schema_template_bd:
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template1
+    bd: ansible_test_multiple_dhcp
+    intersite_bum_traffic: true
+    optimize_wan_bandwidth: false
+    layer2_stretch: true
+    layer2_unknown_unicast: flood
+    layer3_multicast: false
+    unknown_multicast_flooding: flood
+    multi_destination_flooding: drop
+    ipv6_unknown_multicast_flooding: flood
+    arp_flooding: false
+    virtual_mac_address: 00:00:5E:00:01:3C
+    unicast_routing: true
+    subnets:
+    - subnet: 10.0.0.128/24      
+    - subnet: 10.0.1.254/24
+      description: 1234567890
+      primary: true
+    - ip: 192.168.0.254/24
+      description: "My description for a subnet"
+      scope: private
+      shared: false
+      no_default_gateway: true
+      virtual: true
+      primary: true
+    vrf:
+      name: VRF
+      schema: '{{ mso_schema | default("ansible_test") }}'
+      template: Template1
+    dhcp_policies:
+      - name: ansible_test_dhcp_policy1
+        version: 1
+        dhcp_option_policy:
+          name: ansible_test_dhcp_policy_option1
+          version: 1
+      - name: ansible_test_dhcp_policy2
+        version: 1
+        dhcp_option_policy:
+          name: ansible_test_dhcp_policy_option1
+          version: 1
+    state: present
+  register: nm_bd_subnet_second_primary
+  ignore_errors: yes
```

Comment:
Shouldn't this have a WHEN condition?

Reply:
If it is a successive list of tasks that need the same condition you can also use a block statement like: https://github.com/CiscoDevNet/ansible-mso/blob/0ce486646619105c3eb2cbe85762ee72f86fb7a3/tests/integration/targets/mso_user/tasks/main.yml#L30